check_prometheus.py

```diff
@@ -0,0 +1,126 @@
+from __future__ import print_function
+
+import argparse
+import contextlib
+import logging
+import requests
+import sys
+import traceback
+
+METRICS=[]
+
+def status(status, message, force_print=False):
+    global STATUS
+    if status in ('ok', 'warn', 'err'):
+        raise ValueError('The status "%s" is not allowed because it creates a '
+                         'metric called legacy_state' % status)
+    status_line = 'status %s' % status
+    if message is not None:
+        status_line = ' '.join((status_line, str(message)))
+    status_line = status_line.replace('\n', '\\n')
+    STATUS = status_line
+    if force_print:
+        print(STATUS)
+
+def status_ok(message=None, force_print=False, m_name=None):
+    status('okay', message, force_print=force_print)
+
+def status_err(message=None, force_print=False, exception=None, m_name=None):
+    if exception:
+        # a status message cannot exceed 256 characters
+        # 'error ' plus up to 250 from the end of the exception
+        message = message[-250:]
+    status('error', message, force_print=force_print)
+    if exception:
+        raise exception
+    sys.exit(1)
+
+
+def metric(name, metric_type, value, unit=None, m_name=None):
+
+    if len(METRICS) > 49:
+        status_err('Maximum of 50 metrics per check', m_name='maas')
+
+    metric_line = 'metric %s %s %s' % (name, metric_type, value)
+    if unit is not None:
+        metric_line = ' '.join((metric_line, unit))
+
+    metric_line = metric_line.replace('\n', '\\n')
+    METRICS.append(metric_line)
+
+
+def metric_bool(name, success, m_name=None):
+    value = success and 1 or 0
+    metric(name, 'uint32', value, m_name=m_name)
+
+@contextlib.contextmanager
+def print_output():
+    try:
+        yield
+    except SystemExit as e:
+        if STATUS:
+            print(STATUS)
+        raise
+    except Exception as e:
+        logging.exception('The plugin %s has failed with an unhandled '
+                          'exception', sys.argv[0])
+        status_err(traceback.format_exc(), force_print=True, exception=e,
+                   m_name='maas')
+    else:
+        if STATUS:
+            print(STATUS)
+        for metric in METRICS:
+            print(metric)
+
+def check(args):
+    check_name = 'maas_k8s_prometheus_%s' % check
```

Comment:
Seems odd to hard code the "maas_k8s_" here when the script name is more generic. Shouldn't the full string come in from the args?

Reply:
this is meant for multiple checks so only checks vary but that will always be maaas_prometheus